Tasks/DotNetCoreInstallerV0/dotnetcoreinstaller.ts

```diff
@@ -133,22 +200,51 @@ class DotnetCoreInstaller {
 
         // cache tool
         console.log(tl.loc("CachingTool"));
-        let cachedDir = await toolLib.cacheDir(extPath, this.cachedToolName, this.version, this.arch);
-        console.log(tl.loc("SuccessfullyInstalled", this.packageType, this.version));
+        let cachedDir: string;
+        if (globalJson) {            
+            let availableGlobalTool = toolLib.findLocalTool(this.cachedToolName, "0.0.0", this.arch);
+            if (availableGlobalTool == null || availableGlobalTool.length == 0) {
+                // if we never installed any dotnet core sdk.
+                cachedDir = await toolLib.cacheDir(extPath, this.cachedToolName, "0.0.0", this.arch);
+            } else {
+                // here we must do now a little bit magic.
+                // this will replace all existing files. This is only the main dotnet.exe and the eula document.
+                // the other files have there own folders for the sdk version.
+                console.log(tl.loc("OverrideDotnetExe"));
+                console.log(tl.loc("CopySdkFromTo", extPath, availableGlobalTool));
+                tl.cp(extPath + "/", availableGlobalTool, "-rf", true);
```

Comment:
why should the task continue if one version has failed? The down stream tasks will anyways fails due to broken sdk. Shouldn't we cleanup destination and abort task here?

Reply:
This can be happening because you might be trying to copy and replace "dotnet.exe" file. This ca be leading to error,as sometimes the file is being used by dotnet process in the background. 
1. You can try replacing the file with error handling just to be sure if this is causing it.
2. Or you can stop dotnet.exe process before copying, but this can have repercussions if there are multiple agents on the same machine.

Reply:
@paule96 can you please check this as suggested by @hiyadav. If dotnet.exe is the only one which is causing problem, then we can ignore to copy it. And is such a case, you don't need to take workaround mentioned by me [here](https://github.com/Microsoft/azure-pipelines-tasks/pull/9301/files#r253043119). You can then just add a `<version>.Completed` marker file after copying which will indicate that the version is cached successfully and can be used.

Reply:
okay I build this in. But please review the semversion compare logic....

Reply:
close this because no body responde